Kernel/ACPI/Initialize.cpp

```diff
@@ -21,10 +22,15 @@ UNMAP_AFTER_INIT void initialize()
     if (!rsdp.has_value())
         return;
 
-    if (feature_level == AcpiFeatureLevel::Enabled)
-        Parser::initialize<DynamicParser>(rsdp.value());
-    else
-        Parser::initialize<Parser>(rsdp.value());
+    auto facp = StaticParsing::find_table(rsdp.value(), "FACP");
+    if (!facp.has_value())
+        return;
+    auto facp_table = Memory::map_typed<Structures::FADT>(facp.value());
+    u8 irq_line = facp_table->sci_int;
+
+    Parser::initialize(rsdp.value(), facp.value(), irq_line);
+    VERIFY(Parser::the());
```

Comment:
If `initialize` returns void one would assume that it cant fail, but we then VERIFY that it didn't in a round-about way, we should either make `initialize` VERIFY that it didnt initially, or make it return a check-able error code/result etc.